js/packages/teams-ai/src/TurnState.ts

```diff
@@ -57,6 +57,11 @@ export interface DefaultTempState {
      * User authentication tokens
      */
     authTokens: { [key: string]: string };
+
+    /**
+     * Flag indicating whether a token exchange event has already been processed
+     */
+    duplicateTokenExchange?: boolean
```

Comment:
This is used by our library internally. Our library will save the state after processing every request. This will cause ETag conflict when processing the duplicate token exchange activities. So I introduced this flag to let our library be able to skip state saving for duplicated token exchange activity.

I think it can also be adopted by oAuthPrompt. But since we're using the token exchange middleware from botbuilder to handle the duplicate for oAuthPrompt, it's currently only used for TeamsSsoPrompt. I choose the current implementation to reduce the token exchange requests.

Reply:
Since it's used internally in the library we should ideally obscure this from the end user, just like how state the Bot Authentication state is stored. Does this method scale to multiple concurrent login requests from different users?